gen/generate_test.go

```diff
@@ -62,3 +62,51 @@ func TestGenerateWithRelativePaths(t *testing.T) {
 		}
 	}
 }
+
+func TestThriftPackageImporter(t *testing.T) {
+	importer := thriftPackageImporter{
+		ImportPrefix: "github.com/myteam/myservice",
+		ThriftRoot:   "/src/thrift",
+	}
+
+	tests := []struct {
+		File, ServiceName, Relative, Package, ServicePackage string
+	}{
+		{
+			File:           "/src/thrift/foo.thrift",
+			Relative:       "foo",
+			Package:        "github.com/myteam/myservice/foo",
+			ServiceName:    "MyService",
+			ServicePackage: "github.com/myteam/myservice/foo/myservice",
+		},
+		{
+			File:     "/src/thrift/shared/common.thrift",
+			Relative: "shared/common",
+			Package:  "github.com/myteam/myservice/shared/common",
+		},
+	}
+
+	for _, tt := range tests {
+		if tt.Relative != "" {
+			got, err := importer.RelativePackage(tt.File)
+			if assert.NoError(t, err, "RelativePackage(%q)", tt.File) {
+				assert.Equal(t, tt.Relative, got, "RelativePackage(%q)", tt.File)
+			}
+		}
+
+		if tt.Package != "" {
+			got, err := importer.Package(tt.File)
+			if assert.NoError(t, err, "Package(%q)", tt.File) {
+				assert.Equal(t, tt.Package, got, "Package(%q)", tt.File)
+			}
+		}
+
+		if tt.ServicePackage != "" {
```

Comment:
`ServiceName` is the name passed to `ServicePackage`. This is the expected output of calling `ServicePackage(file, name)`.